includes/Customizer.php

```diff
@@ -0,0 +1,593 @@
+				]
+			);
+
+			$wp_customize->add_control(
+				'story-options[stories_archive_label]',
+				[
+					'type'            => 'text',
+					'section'         => self::SECTION_SLUG,
+					'label'           => __( 'Stories archive label', 'web-stories' ),
+					'active_callback' => function() {
+						return ( $this->is_option_enabled( 'show_stories' ) && $this->is_option_enabled( 'show_stories_archive_link' ) );
+					},
+				]
+			);
+
+			$wp_customize->add_setting(
+				'story-options[show_story_poster]',
+				[
+					'default' => $theme_support['show-story-poster-default'],
+					'type'    => 'option',
+				]
+			);
+
+			$wp_customize->add_control(
+				'story-options[show_story_poster]',
+				[
+					'type'            => 'checkbox',
+					'section'         => self::SECTION_SLUG,
+					'label'           => __( 'Show story poster', 'web-stories' ),
+					'active_callback' => function() {
+						return ( $this->is_option_enabled( 'show_stories' ) && $this->is_view_type( 'grid' ) );
+					},
+				]
+			);
+
+		}
+
+	}
+
+	/**
+	 * Gets the view type choices.
+	 *
+	 * @param array $view_type View type to check.
+	 *
+	 * @return array An array of view type choices.
+	 */
+	private function get_view_type_choices( $view_type ) {
+
+		if ( empty( $view_type ) ) {
+			return [ 'circles' => __( 'Circles', 'web-stories' ) ];
+		}
+
+		$view_type = array_keys( $view_type );
+
+		$view_type_choices = [];
+
+		if ( in_array( 'circles', $view_type ) ) {
+			$view_type_choices['circles'] = __( 'Circles', 'web-stories' );
+		}
+
+		if ( in_array( 'grid', $view_type ) ) {
+			$view_type_choices['grid'] = __( 'Grid', 'web-stories' );
+		}
+
+		if ( in_array( 'list', $view_type ) ) {
+			$view_type_choices['list'] = __( 'List', 'web-stories' );
+		}
+
+		if ( in_array( 'carousel', $view_type ) ) {
+			$view_type_choices['carousel'] = __( 'Carousel', 'web-stories' );
+		}
+
+		return $view_type_choices;
+
+	}
+
+
+	/**
+	 * Gets the order choices.
+	 *
+	 * @param array $order An array of order support.
+	 *
+	 * @return array An array of order choices.
+	 */
+	private function get_order_choices( $order ) {
+
+		$order_choices = [];
+
+		if ( empty( $order ) ) {
+			return [
+				'latest'               => __( 'Latest', 'web-stories' ),
+				'oldest'               => __( 'Oldest', 'web-stories' ),
+				'alphabetical'         => __( 'A -> Z', 'web-stories' ),
+				'reverse-alphabetical' => __( 'Z -> A', 'web-stories' ),
+			];
+		}
+
+		$order = array_keys( $order );
+
+		if ( in_array( 'latest', $order ) ) {
+			$order_choices['latest'] = __( 'Latest', 'web-stories' );
+		}
+
+		if ( in_array( 'oldest', $order ) ) {
+			$order_choices['oldest'] = __( 'Oldest', 'web-stories' );
+		}
+
+		if ( in_array( 'alphabetical', $order ) ) {
+			$order_choices['alphabetical'] = __( 'A -> Z', 'web-stories' );
+		}
+
+		if ( in_array( 'reverse-alphabetical', $order ) ) {
+			$order_choices['reverse-alphabetical'] = __( 'Z -> A', 'web-stories' );
+		}
```

Comment:
Could we use `array_intersect_key` here? Something like this ( untested code) ?
```suggestion
		$order_choices = [
				'latest'               => __( 'Latest', 'web-stories' ),
				'oldest'               => __( 'Oldest', 'web-stories' ),
				'alphabetical'         => __( 'A -> Z', 'web-stories' ),
				'reverse-alphabetical' => __( 'Z -> A', 'web-stories' ),
			];

		if ( empty( $order ) ) {
			return $order_choices;
		}

		$order_choices = array_intersect_key( $order_choices, $order );
```